packages/cojson/src/crypto/WasmCrypto.ts

```diff
@@ -54,93 +66,83 @@ export class WasmCrypto extends CryptoProvider<Uint8Array> {
           );
         }
       }),
-    ]).then(([blake3instance]) => new WasmCrypto(blake3instance));
-  }
-
-  randomBytes(length: number): Uint8Array {
-    return randomBytes(length);
+    ]).then(() => new WasmCrypto());
   }
 
   emptyBlake3State(): Uint8Array {
-    return this.blake3Instance.init().save();
+    return blake3_empty_state();
   }
 
   cloneBlake3State(state: Uint8Array): Uint8Array {
-    return this.blake3Instance.load(state).save();
+    return new Uint8Array(state);
   }
 
   blake3HashOnce(data: Uint8Array) {
-    return this.blake3Instance.init().update(data).digest("binary");
+    return blake3_hash_once(data);
   }
 
   blake3HashOnceWithContext(
     data: Uint8Array,
     { context }: { context: Uint8Array },
   ) {
-    return this.blake3Instance
-      .init()
-      .update(context)
-      .update(data)
-      .digest("binary");
+    return blake3_hash_once_with_context(data, context);
   }
 
   blake3IncrementalUpdate(state: Uint8Array, data: Uint8Array): Uint8Array {
-    return this.blake3Instance.load(state).update(data).save();
+    return blake3_update_state(state, data);
   }
 
   blake3DigestForState(state: Uint8Array): Uint8Array {
-    return this.blake3Instance.load(state).digest("binary");
+    return blake3_digest_for_state(state);
   }
 
   newEd25519SigningKey(): Uint8Array {
-    return new Ed25519SigningKey().to_bytes().copyAndDispose();
+    return new_ed25519_signing_key();
   }
 
   getSignerID(secret: SignerSecret): SignerID {
-    return `signer_z${base58.encode(
-      Ed25519SigningKey.from_bytes(
-        new Memory(base58.decode(secret.substring("signerSecret_z".length))),
-      )
-        .public()
-        .to_bytes()
-        .copyAndDispose(),
-    )}`;
+    const signingKey = ed25519_signing_key_from_bytes(
+      base58.decode(secret.substring("signerSecret_z".length)),
+    );
+    const publicKey = ed25519_signing_key_to_public(signingKey);
+    return `signer_z${base58.encode(publicKey)}`;
   }
 
   sign(secret: SignerSecret, message: JsonValue): Signature {
-    const signature = Ed25519SigningKey.from_bytes(
-      new Memory(base58.decode(secret.substring("signerSecret_z".length))),
-    )
-      .sign(new Memory(textEncoder.encode(stableStringify(message))))
-      .to_bytes()
-      .copyAndDispose();
+    const signingKey = ed25519_signing_key_from_bytes(
+      base58.decode(secret.substring("signerSecret_z".length)),
+    );
+    const signature = ed25519_signing_key_sign(
+      signingKey,
+      textEncoder.encode(stableStringify(message)),
+    );
     return `signature_z${base58.encode(signature)}`;
```

Comment:
Can we make this a single sign call to WASM? @bensleveritt 

Same goes for the verify, the less roundtrips we do to WASM memory the better.

I mean putting `ed25519_signing_key_from_bytes`, `ed25519_signing_key_sign` and `base58.encode` behind a single sign call:

```ts
const signature = jazz_ed25519_sign(secret.substring("signerSecret_z".length), stableStringify(message))

return `signature_z${signature}`
```

Reply:
Yeah definitely. I'll just do it in the next PR. I hesitated as the base58 lib isn't wasm, but the fewer boundary movements would indeed be better